pkg/reconciler/kubernetes/initcontroller/initcontroller.go

```diff
@@ -32,9 +32,10 @@ import (
 )
 
 type Controller struct {
-	Manifest         *mf.Manifest
-	Logger           *zap.SugaredLogger
-	VersionConfigMap string
+	Manifest                *mf.Manifest
+	Logger                  *zap.SugaredLogger
+	VersionConfigMap        string
+	IsDashboardSpecReadOnly bool
```

Comment:
this looks a bit hacky and out of place.

let us create a an `PayloadOptions` struct and for now let us add 1 property `ReadOnly`

```bash
type PayloadOptions struct {

    ReadOnly bool
}
```

Then in the `func (ctrl Controller) fetchSourceManifests(ctx context.Context) error`
can become `func (ctrl Controller) fetchSourceManifests(ctx context.Context, opt PayloadOptions) error`

and the 2 calls could become,

`fullaccessManifest, _ := ctrl.InitController(ctx, PayloadOptions{ReadOnly: false})`

and 

`readOnlyManifest, _ := ctrl.InitController(ctx, PayloadOptions{ReadOnly: true})`

Reply:
What would be the readOnly value for tektonPipeline and tektonTriggers ?

Reply:
for other reconcilers, we will not need `readOnlyManifest, _ := ctrl.InitController(ctx, PayloadOptions{ReadOnly: true})` 

there will be only one call, for full access.

Reply:
ie, 
`fullaccessManifest, _ := ctrl.InitController(ctx, PayloadOptions{})`

Reply:
got it thanks will update the pr 👍🏻